src/base/bittorrent/peerinfo.cpp

```diff
@@ -261,116 +260,106 @@ qreal PeerInfo::relevance() const
         m_flags += "S ";
-        flagsDescriptionList += "S = "
-                + tr("peer snubbed");
+        m_flagsDescription += ("S = " + tr("peer snubbed") + '\n');
     }
 
     // I = Peer is an incoming connection
     if (!isLocalConnection()) {
         m_flags += "I ";
-        flagsDescriptionList += "I = "
-                + tr("incoming connection");
+        m_flagsDescription += ("I = " + tr("incoming connection") + '\n');
     }
 
     // K = Peer is unchoking your client, but your client is not interested
     if (!isRemoteChocked() && !isInteresting()) {
         m_flags += "K ";
-        flagsDescriptionList += "K = "
-                + tr("not interested(local) and unchoked(peer)");
+        m_flagsDescription += ("K = "
+            + tr("not interested(local) and unchoked(peer)") + '\n');
     }
 
     // ? = Your client unchoked the peer but the peer is not interested
     if (!isChocked() && !isRemoteInterested()) {
         m_flags += "? ";
-        flagsDescriptionList += "? = "
-                + tr("not interested(peer) and unchoked(local)");
+        m_flagsDescription += ("? = "
+            + tr("not interested(peer) and unchoked(local)") + '\n');
     }
 
     // X = Peer was included in peerlists obtained through Peer Exchange (PEX)
     if (fromPeX()) {
         m_flags += "X ";
-        flagsDescriptionList += "X = "
-                + tr("peer from PEX");
+        m_flagsDescription += ("X = " + tr("peer from PEX") + '\n');
     }
 
     // H = Peer was obtained through DHT
     if (fromDHT()) {
         m_flags += "H ";
-        flagsDescriptionList += "H = "
-                + tr("peer from DHT");
+        m_flagsDescription += ("H = " + tr("peer from DHT") + '\n');
     }
 
     // E = Peer is using Protocol Encryption (all traffic)
     if (isRC4Encrypted()) {
         m_flags += "E ";
-        flagsDescriptionList += "E = "
-                + tr("encrypted traffic");
+        m_flagsDescription += ("E = " + tr("encrypted traffic")  + '\n');
     }
 
     // e = Peer is using Protocol Encryption (handshake)
     if (isPlaintextEncrypted()) {
         m_flags += "e ";
-        flagsDescriptionList += "e = "
-                + tr("encrypted handshake");
+        m_flagsDescription += ("e = " + tr("encrypted handshake") + '\n');
     }
 
     // P = Peer is using uTorrent uTP
     if (useUTPSocket()) {
         m_flags += "P ";
-        flagsDescriptionList += "P = "
-                + QString::fromUtf8(C_UTP);
+        m_flagsDescription += ("P = " + QString::fromUtf8(C_UTP) + '\n');
     }
 
     // L = Peer is local
     if (fromLSD()) {
-        m_flags += 'L';
-        flagsDescriptionList += "L = "
-                + tr("peer from LSD");
+        m_flags += "L ";
+        m_flagsDescription += ("L = " + tr("peer from LSD") + '\n');
     }
+
     m_flags = m_flags.trimmed();
-    m_flagsDescription = flagsDescriptionList.join('\n');
+    m_flagsDescription = m_flagsDescription.trimmed();
```

Comment:
I would reformat it in the following way:
```c++
m_flagsDescription = "first string";
m_flagsDescription += "\n" + "second string";
// ...
m_flagsDescription += "\n" + "Nth string";
```
And `m_flags` too.

Reply:
In this case, the "first string" will be empty and you still need to trim it off in the end... are you sure about it?

Reply:
Oh, right, they're all conditional. Okay, whatever.